test/system/comment_test.rb

```diff
@@ -15,340 +15,246 @@ def setup
-
-    find("p", text: "Reply to this comment...").click()
-
-    # Write a comment
-    page.all("#text-input")[1].set("Great post!")
-
-    page.execute_script <<-JS
-      // Remove first text-input field
-      $("#text-input").remove()
-
-      var $textBox = $("#text-input");
-
-      // Generate fake CTRL + Enter event
-      var press = jQuery.Event("keypress");
-      press.altGraphKey = false;
-      press.altKey = false;
-      press.bubbles = true;
-      press.cancelBubble = false;
-      press.cancelable = true;
-      press.charCode = 10;
-      press.clipboardData = undefined;
-      press.ctrlKey = true;
-      press.currentTarget = $textBox[0];
-      press.defaultPrevented = false;
-      press.detail = 0;
-      press.eventPhase = 2;
-      press.keyCode = 10;
-      press.keyIdentifier = "";
-      press.keyLocation = 0;
-      press.layerX = 0;
-      press.layerY = 0;
-      press.metaKey = false;
-      press.pageX = 0;
-      press.pageY = 0;
-      press.returnValue = true;
-      press.shiftKey = false;
-      press.srcElement = $textBox[0];
-      press.target = $textBox[0];
-      press.type = "keypress";
-      press.view = Window;
-      press.which = 10;
-
-      // Emit fake CTRL + Enter event
-      $textBox.trigger(press);
-    JS
-
-    assert_selector('#comments-list .comment', count: 2)
-    assert_selector('.noty_body', text: 'Comment Added!')
-  end
-
-  test 'comment deletion' do
-    visit "/wiki/wiki-page-path/comments"
-
-    # Create a comment
-    page.execute_script <<-JS
-      var commentForm = $('.comment-form-wrapper')[1];
-      var submitCommentBtn = $(commentForm).find('.btn-primary')[0];
-      var commentTextarea = $(commentForm).find('#text-input')[0]
-
-      $(commentTextarea).val('Great post Jeff!')
-      $(submitCommentBtn).click()
-    JS
-
-    # Delete a comment
-    find('.btn[data-original-title="Delete comment"]', match: :first).click()
-
-    # Click "confirm" on modal
-    page.evaluate_script('document.querySelector(".jconfirm-buttons .btn:first-of-type").click()')
-
-    assert_selector('#comments-list .comment', count: 1)
-    assert_selector('.noty_body', text: 'Comment deleted')
-  end
-
-  test 'formattting toolbar are rendered' do
-    visit "/wiki/wiki-page-path/comments"
-
-    assert_selector('.btn[data-original-title="Bold"]', count: 1)
-    assert_selector('.btn[data-original-title="Italic"]', count: 1)
-    assert_selector('.btn[data-original-title="Header"]', count: 1)
-    assert_selector('.btn[data-original-title="Make a link"]', count: 1)
-    assert_selector('.btn[data-original-title="Upload an image"]', count: 1)
-    assert_selector('.btn[data-original-title="Save"]', count: 1)
-    assert_selector('.btn[data-original-title="Recover"]', count: 1)
-    assert_selector('.btn[data-original-title="Help"]', count: 1)
-
+    comment_response_text = 'wooly woot'
+
+    test "#{page_type_string}: addComment(comment_text)" do
```

Comment:
huh cool! I didn't know you could nest `test` blocks under a loop, but that makes sense as long as the names are unique! Great work!

Reply:
Thanks! 🎈 Creative solution, but it seems to work!